src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_UNMARK = "unmark";
+
+    private static final String MESSAGE_ADD = "Got it. I've added this task:";
+    private static final String MESSAGE_EMPTY_LIST = "Your list is empty.";
+    private static final String MESSAGE_ERROR = "Sorry, I did not understand that.";
+    private static final String MESSAGE_GOODBYE = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_LIST_HEADER = "Here are the tasks in your list:";
+    private static final String MESSAGE_MARK = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_UNMARK = "OK, I've marked this task as not done yet:";
+    private static final String MESSAGE_WELCOME = "Hello! I'm FredBot.\nWhat can I do for you?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initTaskList();
+        showWelcomeMessage();
+        while (true) {
+            String command = readUserInput();
+            executeCommand(command);
+        }
+    }
+
+    private static void initTaskList() {
+        allTasks = new Task[TASK_CAPACITY];
+        count = 0;
+    }
+
+    private static void showWelcomeMessage() {
+        System.out.println(MESSAGE_WELCOME);
+    }
+
+    private static String readUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String input) {
+        final String[] commandWordAndArgs = splitCommandWordAndArgs(input);
+        final String commandWord = commandWordAndArgs[0];
+        final String commandArgs = commandWordAndArgs[1];
+
+        switch (commandWord) {
+            case COMMAND_TODO:
+                executeAddTodo(commandArgs);
+                break;
+            case COMMAND_DEADLINE:
+                executeAddDeadline(commandArgs);
+                break;
+            case COMMAND_EVENT:
+                executeAddEvent(commandArgs);
+                break;
+            case COMMAND_LIST:
+                executeList();
+                break;
+            case COMMAND_MARK:
+                executeMark(commandArgs);
+                break;
+            case COMMAND_UNMARK:
+                executeUnmark(commandArgs);
+                break;
+            case COMMAND_EXIT:
+                executeExit();
+                break;
+            default:
+                executeError();
+                break;
+        }
```

Comment:
The switch case block does not follow the Java coding standard for CS2113 as there should be no indentation for case clauses. (refer to [https://se-education.org/guides/conventions/java/basic.html](url))
Perhaps you can consider configuring your IDE to follow the recommended style to avoid such errors in the future.